pkg/control/control.go

```diff
@@ -488,18 +518,59 @@ func (c *Controller) queryTiDBClusterLogs(dur time.Duration, nodes []clusterType
 				}
 			}
 
-			go func(ns, podName string, nonMatch []string) {
+			// containerName is added to be sure that we only fetch the logs of
+			// the container we want, not included sidecar containers.
+			go func(ns, podName, containerName string, nonMatch []string) {
 				defer wg.Done()
-				texts, err := c.lokiClient.FetchPodLogs(ns, podName,
-					"panic", nonMatch, from, to, false)
+				texts, err := c.lokiClient.FetchPodLogs(ns, podName, containerName,
+					"panic", nonMatch, from, to, 1000, false)
 				if err != nil {
 					log.Infof("failed to fetch logs from loki for pod %s in ns %s", podName, ns)
 				} else if len(texts) > 0 {
 					log.Fatalf("%d panics occurred in ns: %s pod %s. Content: %v", len(texts), ns, podName, texts)
 				}
-			}(n.Namespace, n.PodName, nonMatch)
+			}(n.Namespace, n.PodName, string(n.Component), nonMatch)
+		default:
+		}
+	}
+	wg.Wait()
+}
+
+func (c *Controller) fetchTiDBClusterLogs(nodes []clusterTypes.Node) {
+	wg := &sync.WaitGroup{}
+	to := time.Now()
+
+	for _, n := range nodes {
+		switch n.Component {
+		case clusterTypes.TiDB, clusterTypes.TiKV, clusterTypes.PD:
+			wg.Add(1)
+
+			// containerName is added to be sure that we only fetch the logs of
+			// the container we want, not included sidecar containers.
+			go func(ns, podName, containerName string) {
+				defer wg.Done()
+				texts, err := c.lokiClient.FetchPodLogs(ns, podName, containerName,
+					"", nil, c.lastQueryTime, to, 20000, false)
```

Comment:
Is this enough? 20000 for 4 minutes. We just found one tikv produce 20w logs in 2 minutes when debug is opened @mahjonp

Reply:
Maybe we should reduce the duration of each fetching logs? How about 20000/min?